### PR TITLE
docs: add adoption guide and revamp usage documentation

### PR DESCRIPTION
**Pull request by mjyocca**
## Description

Documentation enhancements for users and partners

### CODE REVIEW

File: docs/ADOPTION.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+| Workflow   |    Terraform CLI (Cloud)            |  TFCI/Terraform Cloud API                      |
+|------------|-------------------------------------|------------------------------------------------|
+| Plan       |  `terraform plan`                   |  commands: `upload`, `run create`              |
+| Apply      |  `terraform apply -auto-approve`    |  commands: `upload`,  `run create`, `run apply`|
+
+#### Terraform Plan
+
+Terraform Cloud CLI can execute a new plan with one command that will upload Terraform configuration and execute a new run in Terraform Cloud.
+
+With tfci and Terraform Cloud API driven runs, these actions are broken up into multiple parts:
```
Comment (aaabdelgany) [inline docs/ADOPTION.md:46]: Q: Should tfci be capitalized throughout this doc?

File: docs/USAGE.md [hunk 2]
```diff
@@ -124,3 +189,27 @@ Push the custom container to your container registry like so
 ```sh
 docker push registry.example.com/namespace/tfci-custom
 ```
+
+## Generating a binary from source
+
+In scenarios where Docker is not available or feasible, you can build a binary directly from the source code.
+
+Prerequisites:
+* git installed
```
Comment (aaabdelgany) [inline docs/USAGE.md:198]: nit: Git

**Review by aaabdelgany — APPROVED** [review]
LGTM, awesome idea.

File: docs/ADOPTION.md (hunk 1 above)
Comment (mjyocca) [reply docs/ADOPTION.md]: Yeah, good eye!

**Review by aaabdelgany — APPROVED** [review]

**Pull request merged**